### PR TITLE
Automatically copy modulemap files into products directory

### LINKED ISSUE #344: module maps should be copied to Products Directory
Currently if a `.modulemap` is in a source folder it will be copied as a resource. It should instead be copied to the Products Directory at the path `include/$(PRODUCT_NAME)`.

There currently is no way to manually specify this either, since `resources` is the only real copy phase supported. Ideally this could be manually specified as well with more `buildPhase`s.

### PR DESCRIPTION
**Pull request by brentleyjones**
Resolves #344.

Depends on #345 (this is actually a small change on top of it).

### CODE REVIEW

**Comment by brentleyjones** [comment]
I’ll add something to the test fixture for this as well.

**Comment by yonaskolb** [comment]
Hi @brentleyjones. Could you please retarget this PR onto the branch it is based off. That makes it easier to actually see the diff 👍

**Comment by brentleyjones** [comment]
@yonaskolb Since both are in my fork, it won’t let me. You can just select the last commit though in Files Changed and review from there.

**Comment by yonaskolb** [comment]
Sure will do. In the future you can create branches in this repo as you’re a collaborator now 😉

**Comment by brentleyjones** [comment]
After digging into this more (#350) we should only copy this over if the target is a static library, since other types don’t need this behavior. Otherwise the file should be ignored (by default). I’ll adjust for this.

File: Sources/XcodeGenKit/SourceGenerator.swift [hunk 1]
```diff
@@ -156,14 +156,20 @@ class SourceGenerator {
...
-    private func getDefaultBuildPhase(for path: Path) -> TargetSource.BuildPhase? {
+    private func getDefaultBuildPhase(for path: Path, targetType: PBXProductType) -> TargetSource.BuildPhase? {
         if path.lastComponent == "Info.plist" {
             return nil
         }
         if let fileExtension = path.extension {
             switch fileExtension {
             case "swift", "m", "mm", "cpp", "c", "cc", "S", "xcdatamodeld": return .sources
             case "h", "hh", "hpp", "ipp", "tpp", "hxx", "def": return .headers
+            case "modulemap":
```
Comment (yonaskolb) [inline Sources/XcodeGenKit/SourceGenerator.swift]: Are there other sorts of files that would have different `CopyFilesSettings`?
If not, then we maybe don't need to pass `PBXProductType` all the way down, and these settings can be provided in `PBXProjGenerator`

Comment (brentleyjones) [reply Sources/XcodeGenKit/SourceGenerator.swift]: Headers for `.staticLibrary` will need to use copy files as well, so we will need to pass the `PBXProductType` down.

**Pull request merged**